libkbfs/root_metadata.go

```diff
@@ -359,13 +359,11 @@ func (md *RootMetadata) AddUnrefBlock(info BlockInfo) {
 
 // AddUpdate adds the newly-updated block to the add block change list.
 func (md *RootMetadata) AddUpdate(oldInfo BlockInfo, newInfo BlockInfo) {
-	if oldInfo.EncodedSize > 0 {
-		md.AddUnrefBytes(uint64(oldInfo.EncodedSize))
-		md.AddRefBytes(uint64(newInfo.EncodedSize))
-		md.AddDiskUsage(uint64(newInfo.EncodedSize))
-		md.SetDiskUsage(md.DiskUsage() - uint64(oldInfo.EncodedSize))
-		md.data.Changes.AddUpdate(oldInfo.BlockPointer, newInfo.BlockPointer)
-	}
+	md.AddUnrefBytes(uint64(oldInfo.EncodedSize))
```

Comment:
I don't remember the original bug we were protecting against here, but I think we've fixed it elsewhere.  We need to allow oldInfos with a 0 `EncodedSize` because newly-created files that haven't been flushed yet will have that property, and we need to make sure the `folderUpdatePrepper` sees the update from the old (randomly-generated) block ID to the final, real block ID.